Taarafo.Core.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Validations.RetrieveById.cs

```diff
@@ -0,0 +1,97 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Moq;
+using Taarafo.Core.Models.Posts;
+using Taarafo.Core.Models.Posts.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Posts
+{
+    public partial class PostServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRetrieveByIdIfIdIsInvalidAndLogItAsync()
+        {
+            // given
+            var invalidPostId = Guid.Empty;
+
+            var invalidPostException =
+                new InvalidPostException();
+
+            invalidPostException.AddData(
+                key: nameof(Post.Id),
+                values: "Id is required");
+
+            var expectedPostValidationException = new
+                PostValidationException(invalidPostException);
+
+            // when
+            ValueTask<Post> retrievePostByIdTask = 
+                this.postService.RetrievePostByIdAsync(invalidPostId);
+
+            // then
+            await Assert.ThrowsAsync<PostValidationException>(() =>
+                retrievePostByIdTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedPostValidationException))),
+                        Times.Once);
+                
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectPostByIdAsync(It.IsAny<Guid>()),
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowNotFoundExceptionOnRetrieveByIdIfPostIsNotFoundAndLogItAsync()
+        {
+            //given
+            Guid someId = Guid.NewGuid();
+            Post emptyPost = null;
+
+            var notFoundPostException =
+                new NotFoundPostException(someId);
+
+            var expectedPostValidationException =
+                new PostValidationException(notFoundPostException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectPostByIdAsync(It.IsAny<Guid>()))
+                    .ReturnsAsync(emptyPost);
+
+            //when
+            ValueTask<Post> retrievePostTask =
```

Comment:
Task here should be called `retrievePostByIdTask` do the same for the rest please.